nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/SystemTasksImpl.java

```diff
@@ -173,15 +174,20 @@ private void setSystemPropertiesFromDomainXml() {
         // 0. server
         // 1. cluster
         // 2. <server>-config or <cluster>-config
-        // 3. domain
+        // 3. deployment-group
+        // 4. domain
         // so we need to add System Properties in *reverse order* to get the
         // right precedence.
 
         List<SystemProperty> domainSPList = domain.getSystemProperty();
         List<SystemProperty> configSPList = getConfigSystemProperties();
         Cluster cluster = server.getCluster();
         List<SystemProperty> clusterSPList = null;
-
+        List<DeploymentGroup> depGroups = server.getDeploymentGroup();
```

Comment:
Can this be null?
Having trouble reproducing atm

Reply:
This cannot be null, but the list may be of length 0.

Reply:
This cannot be null, but is can be a 0-sized list.